### PR TITLE
Add AzureML tutorial

### PR DESCRIPTION
**Pull request by natke**
**Description**: Add a tutorial that shows how to use ONNX Runtime on AzureML

This is an HTML version of the notebook published here, to aid discoverability: https://github.com/microsoft/onnxruntime-inference-examples/blob/main/python/azureml/Inference_Bert_with_OnnxRuntime_on_AzureML.ipynb

Staged here: https://natke.github.io/docs/tutorials/azureml.html

### CODE REVIEW

File: docs/tutorials/azureml.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+```python
+import os
+import logging
+import json
+import numpy as np
+import onnxruntime
+import transformers
+import torch
+
+# The pre process function take a question and a context, and generates the tensor inputs to the model:
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:141]: The pre-process function takes a question and a context, and generates the tensor inputs to the model:

File: docs/tutorials/azureml.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+                                'end_logits': symbolic_names})   # variable length axes
+```
+
+### Run the ONNX model with ONNX Runtime
+
+The following code runs the ONNX model with ONNX Runtime. You can test it locally before deploying it to Azure Machine Learning.
+
+The `init()` function is called at startup, performing the one-off operations such as creating the tokenizer and the ONNX Runtime session.
+
+The `run()` function is called when we run the model using the Azure ML endpoint. Add necessary preprocess() and postprocess() steps.
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:128]: - The `run()` function is called when we run the model using the Azure ML endpoint. 
- Add necessary preprocess() and postprocess() steps.

File: docs/tutorials/azureml.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+
+For local testing and comparison, you can also run the PyTorch model.
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:130]: - For local testing and comparison, you can also run the PyTorch model.

File: docs/tutorials/azureml.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+# - input_ids: the words in the question encoded as integers
+# - attention_mask: not used in this model
+# - token_type_ids: a list of 0s and 1s that distinguish between the words of the question and the words of the context
+# This function also returns the words contained in the question and the context, so that the answer can be decoded into a phrase. 
+def preprocess(question, context):
+    encoded_input = tokenizer(question, context)
+    tokens = tokenizer.convert_ids_to_tokens(encoded_input.input_ids)
+    return (encoded_input.input_ids, encoded_input.attention_mask, encoded_input.token_type_ids, tokens)
+
+# The post process function takes the list of tokens in the question and context, as well as the output of the
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:151]: The post-process function takes the list of tokens in the question and context, the output of the model, the list of log probabilities for the choices of start and end of the answer, and maps it back to an answer to the question that is asked of the context.

File: docs/tutorials/azureml.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    input = "{\"question\": \"What is Dolly Parton's middle name?\", \"context\": \"Dolly Rebecca Parton is an American singer-songwriter\"}"
+
+    run_pytorch(input)
+    print(run(input))
+```
+
+## Deploy model with ONNX Runtime through AzureML
+
+Now that we have the ONNX model and the code to run it with ONNX Runtime, we can deploy it using Azure ML.
+
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:242]: **Component diagram: AzureML deployment with ONNX Runtime (environment, ONNX model, and scoring code)**

File: docs/tutorials/azureml.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+    print("Name:", name,"\tVersion:", m.version, "\tDescription:", m.description, m.tags)
+    
+#     # If you'd like to delete the models from workspace
+#     model_to_delete = Model(ws, name)
+#     model_to_delete.delete()
+```
+
+### Deploy the model and scoring code as an AzureML endpoint
+
+Note that the endpoint interface of the Python SDK has not been publicly released yet, so for this section, we will use the Azure ML CLI.
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:302]: Note that, the endpoint

File: docs/tutorials/azureml.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+print("Torch (ONNX exporter) version: ", torch_version)
+print("Azure SDK version:", azureml.core.VERSION)
+print("ONNX Runtime version: ", onnxruntime.__version__)
+```
+
+### Load your Azure ML workspace
+
+We begin by instantiating a workspace object from the existing workspace created earlier in the configuration notebook.
+
+Note that the following code assumes you have a config.json file containing the subscription information in the same directory as the notebook, or in a sub-directory called .azureml. You can also supply the workspace name, subscription name, and resource group explicity using the Workspace.get() method.
```
Comment (pkreg101) [inline docs/tutorials/azureml.md:264]: Note that, the following code

File: docs/tutorials/azureml.md [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+title: Deploy on AzureML
+descriptions: Deploy high performance question-answer model on AzureML with ONNX Runtime
+parent: Tutorials
+nav_order: 5
+---
+
+# Deploy high performance question-answer model on AzureML with ONNX Runtime
+{: .no_toc }
+
+This tutorial takes a BERT model from HuggingFace, converts it to ONNX, and deploys the ONNX model with ONNX Runtime through AzureML.
```
[the remark this answers is not in the document]
Comment (natke) [reply docs/tutorials/azureml.md:11]: I think BERT is a well known enough term in this domain

File: docs/tutorials/azureml.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+
+In the following sections, we use the HuggingFace Bert model trained with Stanford Question Answering Dataset (SQuAD) dataset as an example. You can also train or fine-tune your own question answer model.
```
[the remark this answers is not in the document]
Comment (natke) [reply docs/tutorials/azureml.md:13]: Ditto

File: docs/tutorials/azureml.md (hunk 4 above)
Comment (natke) [reply docs/tutorials/azureml.md:151]: Reworded

File: docs/tutorials/azureml.md (hunk 5 above)
Comment (natke) [reply docs/tutorials/azureml.md:242]: This is for screen readers so I think the punctuation might stall that

**Review by cassiebreviu — APPROVED** [review]

**Pull request merged**